fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/service/LoanReadPlatformServiceImpl.java

```diff
@@ -1698,9 +1696,32 @@ public LoanTermVariationsData mapRow(final ResultSet rs, @SuppressWarnings("unus
         }
     }
 
+    @Override
+    public Collection<LoanScheduleDelinquencyData> retrieveScheduleDelinquencyData(LocalDate businessLocalDate, List<Long> loanIds) {
+        LoanScheduleDelinquencyByTransactionMapper mapper = new LoanScheduleDelinquencyByTransactionMapper();
+        final StringBuilder sqlBuilder = new StringBuilder(400);
```

Comment:
I don't think this SQL query is complicated enough to use native SQL. Let's go with a JPQL query.

Reply:
Please rewrite it to use JPQL query and Spring Data JPA Projections to get the partial views

Reply:
Done with the use of JPQL + Projections